logstash-core/lib/logstash/config/source_loader.rb

```diff
@@ -0,0 +1,68 @@
+# encoding: utf-8
+require "logstash/config/source/local"
+require "logstash/errors"
+require "thread"
+require "set"
+
+module LogStash module Config
+  class SourceLoader
+    class Loader
+      def initialize(loaders)
+        @loaders = loaders
+      end
+
+      def fetch
+        @loaders.collect(&:pipeline_configs).flatten
+      end
+    end
+
+    include LogStash::Util::Loggable
+
+    def initialize
+      @source_loaders_mutex = Mutex.new
+      @source_loaders = Set.new([LogStash::Config::Source::Local])
+    end
+
+    # This return a ConfigLoader object that will
+    # abstract the call to the different sources and will return multiples pipeline
+    def create(settings)
+      loaders = []
+
+      source_loaders do |config_source|
```

Comment:
it seems this source_loaders synchronized iterator should yield source_loader, but yields `config_source` instead